core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -195,6 +195,27 @@
   public abstract T idleTimeout(long value, TimeUnit unit);
 
   /**
+   * Sets the maximum message size allowed to be received on the channel. If not called,
+   * defaults to 4 MiB. The default provides protection to clients who haven't considered the
+   * possibility of receiving large messages while trying to be large enough to not be hit in normal
+   * usage.
+   *
+   * <p>This method is advisory, and implementations may decide to not enforce this.  Currently,
+   * the only known transport to not enforce this is {@code InProcessTransport}.
+   *
+   * @param max the maximum number of bytes a single message can be.
+   *
+   * @throws IllegalArgumentException if max is negative.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2307")
+  public T maxInboundMessageSize(int max) {
+    @SuppressWarnings("unchecked") // We can't use thisT(), sadly.
+    T self = (T) this;
```

Comment:
The cast looks right. We can just move `thisT()` here and make it protected. It doesn't seem like it being part of the public API would be a problem. Or we make it private if we're super paranoid.